### PR TITLE
Handle FHIR Datatypes

### PR DESCRIPTION
**Pull request by Klochette**
## Description
Handle all FHIR datatypes and create UI components for complex types.

### CODE REVIEW

File: src/state/utils.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          }
+        : {
+            name: splitPath[0],
+            id: rootPath.path,
+            type: rootPath.type,
+            definition: rootPath.definition,
+            children: [],
+            min: rootPath.min,
+            max: rootPath.max
+          };
```
Comment (BPierrick) [inline src/state/utils.ts:54]:
```suggestion
      const newNode = {
            name: splitPath[0],
            id: rootPath.path,
            type: rootPath.type,
            children: [],
            definition: rootPath.definition,
            valueSet: rootPath.valueSet,
            min: rootPath.min,
            max: rootPath.max
          };
```
Why doesn't this work ?

File: src/state/utils.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    const findValueSet = valueSetRequest.data.entry.find(
+                      (
+                        value: FetchedDataCodeSystem
+                      ): FetchedDataCodeSystem | undefined => {
+                        if (value.resource.name === extension.valueString) {
+                          return value;
+                        }
+                        return undefined;
+                      }
+                    );
```
Comment (BPierrick) [inline src/state/utils.ts:102]: This is not how Array.find works.. :/
```suggestion
                    const findValueSet = valueSetRequest.data.entry.find(
                      (
                        value: FetchedDataCodeSystem
                      ) => value.resource.name === extension.valueString);
```

File: src/components/structureDefSettings/utils.ts [hunk 3]
```diff
@@ -29,3 +34,77 @@ export const tooltipValues = {
...
+  });
+  return findPrimitive;
+};
+
+export const createJSONTree = (
+  rootTypes: RenderAttributesTree[],
+  rootArray: any[],
+  primitiveTypes: PrimitiveTypesType[],
+  complexTypes: RenderAttributesTree[]
+) => {
```
[the remark this answers is not in the document]
Comment (Klochette) [reply src/components/structureDefSettings/utils.ts:57]: Yes ! I'd like to !

File: src/state/reducers/fhirDataTypes.ts [hunk 4]
```diff
@@ -42,44 +43,17 @@ export const fhirDataTypes = (
...
         action.payload.primitiveTypes
       );
-
       const structureDefinitionTree = createComplexTypes(
         action.payload.complexTypes,
         action.payload.structureDefinitionTree,
         action.payload.primitiveTypes
       );
-
+      console.log(structureDefinitionTree);
```
Comment (BPierrick) [inline src/state/reducers/fhirDataTypes.ts:56]: To clean

**Review by BPierrick — APPROVED** [review]

**Pull request merged**